dotCMS/src/integration-test/java/com/dotcms/content/elasticsearch/business/ESContentletAPIImplTest.java

```diff
@@ -746,4 +749,77 @@ private static AcknowledgedResponse setClusterAsReadOnly(final boolean value) {
+     * - Create tree contents: A, B, C
+     * - Related content A with B and C
+     * - Related content B with A and C
+     * - Related content C with A and B
+     * Should: All the related content should be saved right
+     */
+    @Test
+    public void selfRelatedContents() throws DotDataException {
+        final Host host = new SiteDataGen().nextPersisted();
+        final ContentType contentType =  new ContentTypeDataGen()
+                .host(host)
+                .nextPersisted();
+
+        final Relationship relationship = new FieldRelationshipDataGen()
+                .child(contentType)
+                .parent(contentType)
+                .nextPersisted();
+
+        Contentlet contentletA = new ContentletDataGen(contentType)
+                .host(host)
+                .nextPersisted();
+
+        Contentlet contentletB = new ContentletDataGen(contentType)
+                .host(host)
+                .nextPersisted();
+
+        Contentlet contentletC = new ContentletDataGen(contentType)
+                .host(host)
+                .nextPersisted();
+
+        contentletA = relateContent(relationship, contentletA, contentletB, contentletC);
+        contentletB = relateContent(relationship, contentletB, contentletA, contentletC);
+        contentletC = relateContent(relationship, contentletC, contentletA, contentletB);
+
+        contentletA.setProperty(relationship.getChildRelationName(), Arrays.asList(contentletB, contentletC));
+
+        ContentletDataGen.checkin(contentletA);
+
+        assertRelatedContents(relationship, contentletA, contentletB, contentletC);
+        assertRelatedContents(relationship, contentletB, contentletA, contentletC);
+        assertRelatedContents(relationship, contentletC, contentletA, contentletB);
+
+    }
+
+    private void assertRelatedContents(Relationship relationship, Contentlet contentletParent,
+            Contentlet... contentsRelated) throws DotDataException {
+        final List<String> contentlets = relationshipAPI
+                .dbRelatedContent(relationship, contentletParent, true)
+                .stream()
+                .map(contentlet -> contentlet.getIdentifier())
+                .collect(Collectors.toList());
+
+        assertEquals(contentsRelated.length, contentlets.size());
+
+        for (Contentlet contentletRelated : contentsRelated) {
+            assertTrue(contentlets.contains(contentletRelated.getIdentifier()));
+        }
+    }
+
+    private Contentlet relateContent(
+            Relationship relationship,
+            Contentlet parentContent,
+            Contentlet... contentletChilds) {
+        final Contentlet parentCheckout = ContentletDataGen.checkout(parentContent);
+        parentCheckout.setProperty(relationship.getChildRelationName(), Arrays.asList(contentletChilds));
+        final Contentlet checkin =  ContentletDataGen.checkin(parentCheckout);
+        return ContentletDataGen.checkout(checkin);
```

Comment:
After discussion this `checkout` should be moved outside of this method

Reply:
done https://github.com/dotCMS/core/pull/20558/commits/ba3d5e566ac3b7b4dc6c03b815a0aa8a748fa7b7